src/modules/qt/filter_gpstext.cpp

```diff
@@ -695,14 +553,15 @@ static mlt_frame filter_process( mlt_filter filter, mlt_frame frame )
 	if ( !dynamic_text || !strcmp( "", dynamic_text ) )
 		return frame;
 
-	mlt_filter text_filter = mlt_properties_get_data( properties, "_text_filter", NULL );
+	mlt_filter text_filter = (mlt_filter)mlt_properties_get_data( properties, "_text_filter", NULL );
 	mlt_properties text_filter_properties = mlt_frame_unique_properties( frame, MLT_FILTER_SERVICE(text_filter) );
 
 	//read and process file if needed
-	process_file(filter);
+	process_file(filter, frame);
+	process_filter_properties(filter, frame);
 
 	// Apply keyword substitution before passing the text to the filter.
-	char* result = calloc( 1, MAX_TEXT_LEN );
+	char* result = (char*)calloc( 1, MAX_TEXT_LEN );
 	substitute_keywords( filter, result, dynamic_text, frame );
 	mlt_properties_set_string( text_filter_properties, "argument", result );
```

Comment:
I know I am late to the party here. But upon inspection, it appears that this filter will not support parallel processing. The "argument" property could be overwritten by subsequent calls to filter_process() before the image is actually processed in another thread.

The string should be copied and pushed on the frame to support parallel processing as is done in the text filter:
https://github.com/mltframework/mlt/blob/master/src/modules/plus/filter_text.c#L166